nevergrad/functions/images/core.py

```diff
@@ -32,36 +56,66 @@ def __init__(self, problem_name: str = "recovering", index: int = 0) -> None:
         # Storing data necessary for the problem at hand.
         assert problem_name == "recovering"  # For the moment we have only this one.
         assert index == 0  # For the moment only 1 target.
-        #path = os.path.dirname(__file__) + "/headrgb_olivier.png"
+        # path = os.path.dirname(__file__) + "/headrgb_olivier.png"
         path = Path(__file__).with_name("headrgb_olivier.png")
         image = PIL.Image.open(path).resize((self.domain_shape[0], self.domain_shape[1]), PIL.Image.ANTIALIAS)
-        self.data = np.asarray(image)[:,:,:3]  # 4th Channel is pointless here, only 255.
+        self.data = np.asarray(image)[:, :, :3]  # 4th Channel is pointless here, only 255.
 
-        # if problem_name == "adversarial": 
-        #     assert index <= 100  # If we have that many target images.
-        #     self.data = ..... (here we load the imagee correspnding to index and problem_name; this is
-        #         # the attacked image.)
-
-        array = ng.p.Array(init=128 * np.ones(self.domain_shape), mutable_sigma=True,)
+        array = ng.p.Array(init=128 * np.ones(self.domain_shape), mutable_sigma=True, )
         array.set_mutation(sigma=35)
         array.set_bounds(lower=0, upper=255.99, method="clipping", full_range_sampling=True)
         max_size = ng.p.Scalar(lower=1, upper=200).set_integer_casting()
         array.set_recombination(ng.p.mutation.Crossover(axis=(0, 1), max_size=max_size)).set_name("")  # type: ignore
+
         super().__init__(self._loss, array)
         self.register_initialization(problem_name=problem_name, index=index)
         self._descriptors.update(problem_name=problem_name, index=index)
 
     def _loss(self, x: np.ndarray) -> float:
+        assert self.problem_name == "recovering"
         x = np.array(x, copy=False).ravel()
         x = x.reshape(self.domain_shape)
         assert x.shape == self.domain_shape, f"Shape = {x.shape} vs {self.domain_shape}"
-
-
         # Define the loss, in case of recovering: the goal is to find the target image.
-        assert self.problem_name == "recovering"
         assert self.index == 0
         value = np.sum(np.fabs(np.subtract(x, self.data)))
-
-        # Here we should implement "adversarial" and others.
         return value
 
+
+class ImageAdversarial(base.ExperimentFunction):
+    def __init__(self, params: dict = {}) -> None:
```

Comment:
this needs a simplified signature, with maybe a couple of parameters, like `target: str, classifier:str, epsilon: float`, and depending on the names, you instantiate the corresponding classifier etc. Otherwise this wont be useable out of the box, and hard to use in benchmarks

Reply:
I agree with this. Or everything has a default value.
One of the arguments could be the index in ImageNet, with default value 0. The code should look for the default path in /datasetxxxxx/xxxx, use those images if they are there --- otherwise we can use one or two images (we will not copy the entire image net inside Nevergrad :-) ).

(we could also do something like https://stackoverflow.com/questions/40251150/using-python-to-download-images-via-urllib-and-testing-if-the-website-exists, maybe, but maybe not for now...)